packages/electron-builder/src/targets/appx.ts

```diff
@@ -44,38 +44,60 @@ export default class AppXTarget extends Target {
     const preAppx = path.join(this.outDir, `pre-appx-${getArchSuffix(arch)}`)
     await emptyDir(preAppx)
 
+    const destination = path.join(this.outDir, packager.expandArtifactNamePattern(this.options, "appx", arch))
     const vendorPath = await getSignVendorPath()
 
     const templatePath = path.join(__dirname, "..", "..", "templates", "appx")
     const safeName = sanitizeFileName(appInfo.name)
-    const resourceList = await packager.resourceList
-    await BluebirdPromise.all([
-      BluebirdPromise.map(["44x44", "50x50", "150x150", "310x150"], size => {
-        const target = path.join(preAppx, "assets", `${safeName}.${size}.png`)
-        if (resourceList.includes(`${size}.png`)) {
-          return copyFile(path.join(packager.buildResourcesDir, `${size}.png`), target)
-        }
-        return copyFile(path.join(vendorPath, "appxAssets", `SampleAppx.${size}.png`), target)
-      }),
-      copyDir(appOutDir, path.join(preAppx, "app")),
-      this.writeManifest(templatePath, preAppx, safeName, arch, publisher)
-    ])
+    
+    const customAssetsFolder = this.options.assetsFolder
+    if (customAssetsFolder) {
+      const customAssetsPath = path.resolve(this.packager.projectDir, customAssetsFolder)
+      await BluebirdPromise.all([
+        copyDir(customAssetsPath, path.join(preAppx, 'assets')),
+        copyDir(appOutDir, path.join(preAppx, "app")),
+      ])
+    } else {
+      const resourceList = await packager.resourceList
+      await BluebirdPromise.all([
+        BluebirdPromise.map(["44x44", "50x50", "150x150", "310x150"], size => {
+          const target = path.join(preAppx, "assets", `${safeName}.${size}.png`)
+          if (resourceList.includes(`${size}.png`)) {
+            return copyFile(path.join(packager.buildResourcesDir, `${size}.png`), target)
+          }
+          return copyFile(path.join(vendorPath, "appxAssets", `SampleAppx.${size}.png`), target)
+        }),
+        copyDir(appOutDir, path.join(preAppx, "app")),
+      ])
+    }
 
-    const destination = path.join(this.outDir, packager.expandArtifactNamePattern(this.options, "appx", arch))
-    const args = ["pack", "/o", "/d", preAppx, "/p", destination]
-    use(this.options.makeappxArgs, (it: Array<string>) => args.push(...it))
+    this.writeManifest(templatePath, preAppx, safeName, arch, publisher, this.options.assetNames)
```

Comment:
If custom dir specified, we should populate missed values in the asset names. Right? Otherwise strange — why we require explicit configuration of names.